### PR TITLE
Add types

### PR DESCRIPTION
**Pull request by pd4d10**

### CODE REVIEW

**Comment by pd4d10** [comment]
https://travis-ci.org/github/remarkjs/remark-external-links/jobs/718949083#L235

```
  types/index.d.ts:3:22
  ✖  3:22  unified should be listed in the project's dependencies. Run npm i -S unified to add it  import/no-extraneous-dependencies
```

Shall we add it to dependencies or disable this rule?

**Comment by ChristianMurphy** [comment]
> Shall we add it to dependencies or disable this rule?

I'll defer to @wooorm on this one, IIRC the current preference is to disable this rule.

**Comment by pd4d10** [comment]
Another question, should `RemarkExternalLinksOptions` be exposed to users?

**Comment by ChristianMurphy** [comment]
> Another question, should `RemarkExternalLinksOptions` be exposed to users?

Yes, unified/remark plugins often depend on each other, having options reusable allows reuse.

**Review by wooorm — APPROVED** [review]
lgtm, with `RemarkExternalLinksOptions`!

**Review by ChristianMurphy — APPROVED** [review]

**Comment by pd4d10** [comment]
> Another question, should `RemarkExternalLinksOptions` be exposed to users?

Currently we use `export = remarkExternalLinks` and could not do something like `export { RemarkExternalLinksOptions }` any more. Is there any best practice for this case?

**Comment by ChristianMurphy** [comment]
There is,
`export =` is needed  to ensure that the types work when https://www.typescriptlang.org/tsconfig#allowSyntheticDefaultImports is `false`.
to allow an interface to also be included use a namespace that matches the export.

For example https://github.com/remarkjs/remark-frontmatter/blob/main/types/index.d.ts#L5 and https://github.com/remarkjs/remark-frontmatter/blob/3414a84ef27ead52701cae985994f67f17b414da/types/index.d.ts#L61-L63

**Comment by pd4d10** [comment]
@ChristianMurphy Get it. Updated

**Review by ChristianMurphy — APPROVED** [review]
thanks @pd4d10! :bow:

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+     * [node]: https://github.com/syntax-tree/hast#nodes
+     * [children]: https://github.com/syntax-tree/unist#child
+     * [g201]: https://www.w3.org/WAI/WCAG21/Techniques/general/G201
+     */
+    content?: Node | Node[]
+    /**
+     * [`Properties`][properties] to add to the `span` wrapping `content`, when
+     * given.
+     */
+    contentProperties?: Record<string, string>
```
Comment (wooorm) [inline types/index.d.ts:54]: Whoops! `Properties` is not just `<string, string>`. There can be other values as well: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/73e5ee37847e0ad313459222642db3eed1e985b7/types/hast/index.d.ts#L72-L77

Comment (pd4d10) [reply types/index.d.ts]: Updated

**Review by BarryThePenguin — APPROVED** [review]

**Review by wooorm — APPROVED** [review]

**Pull request merged**